src/georaster-layer-for-leaflet.ts

```diff
@@ -0,0 +1,592 @@
+            return result;
+          }, [])
+      );
+    }
+  },
+
+  createTile: function (coords: Coords, done: DoneCallback) {
+    /* This tile is the square piece of the Leaflet map that we draw on */
+    const tile = L.DomUtil.create("canvas", "leaflet-tile") as HTMLCanvasElement;
+    tile.height = this.tileHeight;
+    tile.width = this.tileWidth;
+    const context = tile.getContext("2d");
+
+    return this.drawTile({ tile, coords, context, done });
+  },
+
+  drawTile: function ({ tile, coords, context, done }: DrawTileOptions) {
+    let error: Error;
+
+    const inSimpleCRS = this.getMap().options.crs === L.CRS.Simple;
+
+    // Unpacking values for increased speed
+    const { rasters, xmin, ymax } = this;
+    const rasterHeight = this.height;
+    const rasterWidth = this.width;
+
+    const pixelHeight = inSimpleCRS ? this.getBounds()._northEast.lat / rasterHeight : this.pixelHeight;
+    const pixelWidth = inSimpleCRS ? this.getBounds()._northEast.lng / rasterWidth : this.pixelWidth;
+
+    // these values are used, so we don't try to sample outside of the raster
+    const { xMinOfLayer, xMaxOfLayer, yMinOfLayer, yMaxOfLayer } = this;
+    const boundsOfTile = this._tileCoordsToBounds(coords);
+
+    const xMinOfTileInMapCRS = boundsOfTile.getWest();
+    const xMaxOfTileInMapCRS = boundsOfTile.getEast();
+    const yMinOfTileInMapCRS = boundsOfTile.getSouth();
+    const yMaxOfTileInMapCRS = boundsOfTile.getNorth();
+
+    let rasterPixelsAcross, rasterPixelsDown;
+    if (inSimpleCRS || this.projection === EPSG4326) {
+      // width of the Leaflet tile in number of pixels from original raster
+      rasterPixelsAcross = Math.ceil((xMaxOfTileInMapCRS - xMinOfTileInMapCRS) / pixelWidth);
+      rasterPixelsDown = Math.ceil((yMaxOfTileInMapCRS - yMinOfTileInMapCRS) / pixelHeight);
+      // note: why do we check for getProjector() - SFR 2021-01-20
+    } else if (this.getProjector()) {
+      const projector = this.getProjector();
+      // convert extent of Leaflet tile to projection of the georaster
+      const topLeft = projector.inverse({ x: xMinOfTileInMapCRS, y: yMaxOfTileInMapCRS });
+      const topRight = projector.inverse({ x: xMaxOfTileInMapCRS, y: yMaxOfTileInMapCRS });
+      const bottomLeft = projector.inverse({ x: xMinOfTileInMapCRS, y: yMinOfTileInMapCRS });
+      const bottomRight = projector.inverse({ x: xMaxOfTileInMapCRS, y: yMinOfTileInMapCRS });
+
+      rasterPixelsAcross = Math.ceil(Math.max(topRight.x - topLeft.x, bottomRight.x - bottomLeft.x) / pixelWidth);
+      rasterPixelsDown = Math.ceil(Math.max(topLeft.y - bottomLeft.y, topRight.y - bottomRight.y) / pixelHeight);
+    }
+
+    const { resolution } = this.options;
+
+    // prevent sampling more times than number of pixels to display
+    const numberOfSamplesAcross = Math.min(resolution, rasterPixelsAcross);
+    const numberOfSamplesDown = Math.min(resolution, rasterPixelsDown);
```

Comment:
@DanielJDufour, the compiler complains that `rasterPixelsAcross` & `rasterPixelsDown` could be undefined as they are set within the `if` block (lns: 237-252). `Math.min` expects a type number only. Is it possible to give the `rasterPixels*` variables a default of 0? Or to move lines 257&258 within the `if` block?

Reply:
Yes, you could definitely default rasterPixelsAcross to zero.  Or you could default it to 256 (which is the standard map tile size).  Either way.

Reply:
Great. And I can do the same for `rasterPixelsDown`?

Reply:
Seems like you solved this, so I'll resolve this thread.